### PR TITLE
Update .travis.yml

### PR DESCRIPTION
**Pull request by Flamefire**
CodeCov.io added support for settings via `codecov.yml`
This also adds coveralls statistics as codecov is a bit buggy ATM

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,5 +1,8 @@
 sudo: false
 language: cpp
+branches:
+    only:
+        - master
```
Comment (Flow86) [inline .travis.yml]: that will break pull request checks

Comment (Flamefire) [inline .travis.yml]: It doesn't. PRs are handled additionally. Have a look at the first commit in this PR, that did not have this. There are 2 builds from travis, which should not be.

**Comment by Flamefire** [comment]
CodeCov seems to have some issues at the moment. I'd like to add support for coveralls as well, to compare those 2 services and make sure, there is at least 1 working ;)

**Comment by Flow86** [comment]
coveralls is activated now, you only have to write the .coveralls.yml

**Comment by Flamefire** [comment]
The yml file for coveralls is only needed for private repos. Works now and looks good. Although I liked the old design of codecov better. Let's see how it looks like once their issues are resolved.

**Comment by stevepeak** [comment]
@Flamefire you can remove the "changes" above the diff if you desire

```
comment:
  layout: "header, diff"  # old-style
```

**Comment by Flamefire** [comment]
@stevepeak I don't mind having them. They might be helpful (if they only were correct -> https://github.com/codecov/support/issues/164)

**Pull request merged**